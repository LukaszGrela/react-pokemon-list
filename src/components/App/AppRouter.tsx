import React, { ReactNode } from 'react';
import { ConnectedRouter } from 'connected-react-router';
import { Switch, Route, Redirect } from 'react-router-dom';
import { history } from '../../store/root-reducer';
import Home from '../Home/Home';

const AppRouter: React.FC = (): JSX.Element => {
  return (
    <ConnectedRouter history={history}>
      <Switch>
        <Route
          exact
          path='/index.html'
          render={(): ReactNode => {
            return <Redirect to='/1' />;
          }}
        />
        <Route
          exact
          path='/'
          render={(): ReactNode => {
            return <Redirect to='/1' />;
          }}
        />
        <Route
          path='/:page'
          exact
          render={(): ReactNode => {
            return <Home />;
          }}
        />
      </Switch>
    </ConnectedRouter>
  );
};

export default AppRouter;
