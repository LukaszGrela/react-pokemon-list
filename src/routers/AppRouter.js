import React from 'react';
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';

import MediaQuery from 'react-responsive';

import Page404 from '../pages/Page404';
import Home from '../pages/Home';
import PokemonDetailsPage from '../pages/PokemonDetailsPage';

const AppRouter = () => (
    <BrowserRouter>
        <div>
            <Switch>
                <Route exact path="/index.html" component={() => {
                    return <Redirect to='/' />
                }} />
            </Switch>
            <MediaQuery
                key={'media-query'}
                minWidth={410}>
                {
                    (matches) => {
                        if (matches) {
                            /* normal */
                            return <Switch><Route component={Home} key={'route-1'} /></Switch>
                        } else {
                            /* small devices */
                            return <Switch>
                                <Route path="/" exact component={Home} key={'route-1'} />
                                <Route path="/pokemon/:id" exact component={PokemonDetailsPage} key={'route-2'} />
                                <Route component={Page404} key={'route-3'} />
                            </Switch>
                        }
                    }
                }
            </MediaQuery>
        </div>
    </BrowserRouter>
);

export default AppRouter;