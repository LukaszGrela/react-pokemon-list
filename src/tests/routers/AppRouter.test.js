import React from 'react';

import { shallow } from 'enzyme';
import AppRouter from '../../routers/AppRouter';

test('Should render AppRouter correctly', () => {
    const wrapper = shallow(<AppRouter />);

    expect(wrapper).toMatchSnapshot();
});