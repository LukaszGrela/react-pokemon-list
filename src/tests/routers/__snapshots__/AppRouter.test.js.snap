// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AppRouter correctly 1`] = `
<BrowserRouter>
  <div>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/index.html"
      />
    </Switch>
    <MediaQuery
      key="media-query"
      minWidth={410}
      values={Object {}}
    />
  </div>
</BrowserRouter>
`;
